File: backend/app/agent/llm.py
import os

from config import env

from langchain_openai import ChatOpenAI
from langchain_litellm import ChatLiteLLM
import litellm

os.environ["BRAINTRUST_API_KEY"] = env.BRAINTRUST_API_KEY #type: ignore
litellm.callbacks = ["braintrust"]

if env.OPENAI_API_KEY:
    llm = ChatOpenAI(api_key=env.OPENAI_API_KEY, model="gpt-4.1") #type: ignore
elif env.LITELLM_API_KEY:
    llm = ChatLiteLLM(api_key=env.LITELLM_API_KEY, api_base="https://llm.keyvalue.systems", model="litellm_proxy/gpt-4.1")

File: backend-ts/src/agent/llm.ts
import { ChatOpenAI } from "@langchain/openai";
import { ChatGroq } from "@langchain/groq";

// If you have a direct OpenAI API Key
// export const llm = new ChatOpenAI({
//     openAIApiKey: process.env.OPENAI_API_KEY,
//     modelName: "gpt-4.1" // or whatever model you use
// });

// Since the Python code used LiteLLM pointing to a custom proxy,
// we use ChatOpenAI in TypeScript but override the baseUrl to point to LiteLLM.
// LiteLLM is fully compatible with the OpenAI API format!
// export const llm = new ChatOpenAI({
//   apiKey: process.env.LITELLM_API_KEY || "dummy",
//   modelName: "litellm_proxy/gpt-5",
//   configuration: {
//     baseURL: "https://llm.keyvalue.systems", // Assuming this is your LiteLLM proxy URL
//   },
// });


export const llm = new ChatGroq({
  apiKey: process.env.GROQ_API_KEY,
  model: "openai/gpt-oss-120b",
  temperature: 0,
});